core/src/avm2/globals/flash/net/socket.rs

```diff
@@ -0,0 +1,306 @@
+use crate::avm2::bytearray::Endian;
+use crate::avm2::error::make_error_2008;
+pub use crate::avm2::object::socket_allocator;
+use crate::avm2::{Activation, Error, Object, TObject, Value};
+use crate::context::UpdateContext;
+
+pub fn connect<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let socket = match this.as_socket() {
+        Some(socket) => socket,
+        None => return Ok(Value::Undefined),
+    };
+
+    let host = match args.get(0) {
```

Comment:
This should be `let host = args.get_value(0);`, since the first parameter is guaranteed.